src/crypto/keccak.h

```diff
@@ -15,6 +15,17 @@
 #define ROTL64(x, y) (((x) << (y)) | ((x) >> (64 - (y))))
 #endif
 
+// SHA3 Algorithm context.
+typedef struct KECCAK_CTX
+{
+  // 1600 bits algorithm hashing state
+  uint64_t hash[25];
+  // 1536-bit buffer for leftovers
+  uint64_t message[24];
```

Comment:
That seems too large. Looks like it needs 136 bytes, so 17 uint64_t. Is this wrong ?